### PR TITLE
Configurable escaping

### PR DESCRIPTION
**Pull request by wattsm**
This PR:

- Converts the project structure to Leiningen for ease of development/deployment.
- Makes the quote escape character configurable via the `:escape` option (defaulting to the standard CSV option of `"`)

### CODE REVIEW

File: src/clojure/data/csv.clj [hunk 1]
```diff
@@ -129,18 +147,21 @@
 
    Valid options are
      :separator (Default \\,)
+     :escape (A charater used to escape quotes inside cells. Defaults to \\\".)
```
Comment (AdamClements) [inline src/clojure/data/csv.clj]: sp

File: src/clojure/data/csv.clj [hunk 2]
```diff
@@ -129,18 +147,21 @@
...
   (let [opts (apply hash-map options)
         separator (or (:separator opts) \,)
+        escape (or (:escape opts) \")
         quote (or (:quote opts) \")
         quote? (or (:quote? opts) #(some #{separator quote \return \newline} %))
         newline (or (:newline opts) :lf)]
     (write-csv* writer
 		data
 		separator
+    escape
```
Comment (AdamClements) [inline src/clojure/data/csv.clj]: alignment

**Review by AdamClements — APPROVED** [review]

**Comment by AdamClements** [comment]
Can't merge as it's draft, but otherwise fine. I'm slightly flip flopping on we should make an evahealth clojars account

**Comment by wattsm** [comment]
> Can't merge as it's draft, but otherwise fine. I'm slightly flip flopping on we should make an evahealth clojars account

I could always create one with a generic `@evahealth.co.uk` email address and stick the password into 1Password?

**Pull request merged**